### PR TITLE
AMP-32: Add stylelint for AMP

### PR DESCRIPTION
**Pull request by cole-sanderson**
Add `stylelint` to work with disallowed/restricted AMP styles. `stylelint` will only point to sharing styles with PWA/AMP: base, animations, lib and utilities. 

 **JIRA**: https://mobify.atlassian.net/browse/AMP-32
 **Linked PRs**: N/A

## Changes
- Add `stylelint` to `lint`

## How to test-drive this PR
- Run `npm install`
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Add error styles in base, animations, lib, or utilities file. Error: `!important`, `i-amp-` tag name, `-amp-` class, property: `behavior`, `-moz-binding`, `filter`, or any transition declaration that's not `opacity` or `transform`.

### CODE REVIEW

File: amp/package.json [hunk 1]
```diff
@@ -57,10 +58,11 @@
     "dev:serve": "nodemon build/main.js",
     "dev:watch": "webpack --watch",
     "dev": "npm-run-all --parallel dev:watch dev:serve",
-    "lint": "npm run lint:js && npm run lint:sass",
+    "lint": "npm run lint:js && npm run lint:sass && npm run lint:amp-scss",
     "lint:fix": "npm run lint:js -- --fix",
     "lint:js": "eslint '**/*.{js,jsx}'",
-    "lint:sass": "sass-lint -c .sass-lint.yml 'app/**/*.scss' -v",
+    "lint:sass": "sass-lint -c ../web/.sass-lint.yml 'app/**/*.scss' -v",
```
Comment (jeffkamo) [inline amp/package.json]: Is CI setup to lint things in the amp directory yet? It looks like CI is passing, but there are currently the following errors:

screen shot 2017-05-31 at 10 56 41 am

Comment (cole-sanderson) [inline amp/package.json]: Just pushed to fix lint errors. It was errors in base and theme files in amp-lightbox

Comment (kschmidtdev) [inline amp/package.json]: CI isn't set up yet but soooooon!

**Comment by olibrook** [comment]
The rules look good to me, but this makes the `web` project dependent on the `amp` project. Can you suggest somewhere to put common assets like this? @cole-sanderson @jeffkamo 

It sounds like a requirement is going to be coming very soon that we should be able to delete the `amp` or `web` folder without breaking the other project.

**Comment by jeffkamo** [comment]
^^^ I made the above suggestion in light of the fact that at this time, we don't have a place to put things that are shared across the three different apps. At the very least, dependencies should certainly be handled in a way that makes it simple to reduce the repo down to just the web.

**Comment by cole-sanderson** [comment]
Just moved .stylelint.yml into web from amp for now. Later when we have "common" folder set up and we can fix this issue.

**Comment by olibrook** [comment]
Since amp already depends on the web project at the moment, that's a thumbs-up from me.

👍

**Pull request merged**